modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -380,18 +424,53 @@ export default class ParticipantConnectionStatus {
             isConnActiveByJvb = true;
         }
 
-        const isConnectionActive
-            = isConnActiveByJvb
-                && (isVideoMuted || (isInLastN && !isVideoTrackFrozen));
+        let newState = ParticipantConnectionStatus.INACTIVE;
+
+        if (isConnActiveByJvb) {
+            if (isInLastN) {
+                if (isVideoTrackFrozen) {
+                    // if participant was inactive, and is in lastN,
+                    // it is restoring
+                    const isInactive = participant.getConnectionStatus()
+                        === ParticipantConnectionStatus.INACTIVE;
+
+                    newState = isInactive
```

Comment:
well restoring can be something like:
RESTORING = isInLastN & isVideoTrackFrozen & timeSinceEneteredLastN < 5000;
else INTERRUPTED
No ?
This is a simplification of course I need to think this throughly, but it is possible to figure out restoring without previous state (the timestamp is kind of tracking the previous state).

Also it's important that isVideoTrackFrozen can be true only on Chrome, so we'd have to think about that some more (not sure if that changes anything here).